### ISSUE TITLE
max_grad_norm doesn't work well and got "!!!!!!!!!!!!!!" output

### ISSUE THREAD

**Issue by kxfan2002** [hiyouga/EasyR1#177]
Hi, and thanks for your great work on this project!

I'm encountering an issue while training with `max_grad_norm: 1.0`. Initially, the training process appears normal (see images), but after some iterations, the model starts outputting only "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!".

I suspect this issue might be caused by an outlier value(like grad_norm) affecting the training dynamics. Could this be related to a missing or incorrect setting in my configuration?

![Image](https://github.com/user-attachments/assets/66cda5ed-220c-4c25-8a1a-08d9827fe1dc)

![Image](https://github.com/user-attachments/assets/9b6cd911-68e1-4e29-b035-7a11bf55a171)

Here is my full .yaml configuration:
```
data:
  train_files: OCR
  val_files: demo2
  prompt_key: prompt
  answer_key: answer
  image_key: images
  rollout_batch_size: 128
  max_prompt_length: 8192
  max_response_length: 2048
  use_addition_column: True
  shuffle: true
  seed: 1
  max_pixels: 4194304
  min_pixels: 262144

algorithm:
  adv_estimator: grpo
  kl_coef: 0.0001

worker:
  actor:
    global_batch_size: 32
    micro_batch_size_per_device_for_update: 4
    micro_batch_size_per_device_for_experience: 8
    max_grad_norm: 1.0
    entropy_coeff: 1.0e-3
    use_kl_loss: true
    kl_loss_coef: 1.0e-2
    kl_loss_type: low_var_kl
    padding_free: true
    ulysses_sequence_parallel_size: 1
    model:
      model_path: Qwen/Qwen2.5-7B-Instruct
      enable_gradient_checkpointing: true
      trust_remote_code: false
      freeze_vision_tower: false
    optim:
      lr: 1.0e-6
      weight_decay: 1.0e-2
      lr_warmup_ratio: 0.0
    offload:
      offload_params: true
      offload_optimizer: true
    fsdp:
      enable_full_shard: true
      enable_cpu_offload: false
      enable_rank0_init: true

  rollout:
    temperature: 1.0
    tensor_parallel_size: 2
    gpu_memory_utilization: 0.4
    n: 5
    enable_chunked_prefill: true
    enforce_eager: false
    limit_images: 0
    max_num_batched_tokens: 12000

  ref:
    offload:
      offload_params: false
    fsdp:
      enable_full_shard: true
      enable_cpu_offload: true
      enable_rank0_init: true

  reward:
    reward_type: function 
    compute_score: choice
    val_compute_score: math

trainer:
  total_episodes: 3
  logger: ["console", "wandb"]
  project_name: easy_r1
  experiment_name: qwen2_5_7b_bs
  n_gpus_per_node: 8
  nnodes: 1
  val_freq: -1
  val_before_train: false
  val_only: false
  val_generations_to_log: 1
  save_freq: 500
  remove_previous_ckpt: false
  remove_ckpt_after_load: false
  save_checkpoint_path: null
  load_checkpoint_path: null
```
Any insights or suggestions would be greatly appreciated. Thank you for your help!

**Comment by Artessay** [comment]
I’ve run into almost the exact same problem as you. After training for a few steps, the model's performance suddenly plummets—it starts ignoring the required output format and sometimes just spits out nothing but exclamation marks. Waiting for suggestions to solve this issue.

**Comment by Artessay** [comment]
Why this issue was closed without any comment?

**Comment by ruiiu** [comment]
same here, any reason?

**Comment by kxfan2002** [comment]
> same here, any reason?

Hi! I still do not know the exact reason, but I would like to share my personal experience with this issue.
My training code had been running successfully before. However, one day I started getting NaN values for the loss from the second training step (model response !!!!!!). After switching to a different GPU node, the problem disappeared.
I have heard that there might be some precision issues when using the A800 (sorry, I do not have more detailed information about this).

Hope this helps, and I wish you a quick resolution!
